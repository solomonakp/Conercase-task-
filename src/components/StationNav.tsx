import React from 'react';
import IconButton from './IconButton';
import styles from './styles/StationNav.module.scss';
import back from '../assets/back-arrow.png';
import power from '../assets/switch.png';

export const StationNav = () => {
  return (
    <div className={styles.stationNav}>
      <nav>
        <ul>
          <li>
            <IconButton
              src={back}
              testId='nav-btn'
              alt='back-icon-btn'
              label='back button'
            />
          </li>
          <li>Stations</li>
          <li>
            <IconButton
              src={power}
              testId='nav-btn'
              alt='power-icon-btn'
              label='power button'
            />
          </li>
        </ul>
      </nav>
    </div>
  );
};
