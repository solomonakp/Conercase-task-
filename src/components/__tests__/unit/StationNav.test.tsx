import React from 'react';
import { axe } from 'jest-axe';
import { render, fireEvent } from '@testing-library/react';
import { StationNav } from '../../StationNav';

describe('The <StationNav/> component', () => {
  const setUpHeader = () => render(<StationNav />);

  it('renders correctly', () => {
    const { asFragment } = setUpHeader();

    expect(asFragment()).toMatchSnapshot();
  });

  it('has  two icon Buttons', () => {
    const { getAllByTestId } = setUpHeader();

    const iconButton = getAllByTestId('nav-btn');

    expect(iconButton).toHaveLength(2);
  });
  it('button should be clickable', () => {
    const { getByAltText } = setUpHeader();

    console.log = jest.fn();

    const iconButton = getByAltText('back-icon-btn');

    fireEvent.click(iconButton);

    expect(console.log).toHaveBeenCalled();
  });
  it('❌ has no accessibility validation', async () => {
    const { container } = setUpHeader();

    const result = await axe(container);

    expect(result).toHaveNoViolations();
  });
});
