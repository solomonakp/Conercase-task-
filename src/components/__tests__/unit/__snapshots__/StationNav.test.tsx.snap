// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <StationNav/> component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="stationNav"
  >
    <nav>
      <ul>
        <li>
          <button
            aria-label="back button"
            class="iconButton"
            data-testid="nav-btn"
            style="width: 20px; height: 20px;"
          >
            <img
              alt="back-icon-btn"
              src="back-arrow.png"
            />
          </button>
        </li>
        <li>
          Stations
        </li>
        <li>
          <button
            aria-label="power button"
            class="iconButton"
            data-testid="nav-btn"
            style="width: 20px; height: 20px;"
          >
            <img
              alt="power-icon-btn"
              src="switch.png"
            />
          </button>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
